### ISSUE TITLE
Entity IDs changing when applying snapshot?

### ISSUE THREAD

**Issue by frnsys** [hankjordan/bevy_save#21]
Hi, I'm struggling with an issue I can't quite figure out.

I'm building a snapshot like so:

```
    for ent in &entities {
        let name = world.get::<Name>(*ent);
        println!("{:?} name {:?}", ent, name);
    }

    Snapshot::builder(world)
        .extract_entities(entities.into_iter())
        .build()
```

Then applying it like so:

```
            let filter = <dyn Filter>::new::<With<Selectable>>();
            snapshot
                .applier(world)
                .despawn(bevy_save::DespawnMode::MissingWith(Box::new(filter)))
                .hook(move |entity, cmds| {
                    println!("---------spawning entity {:?}", entity.id());
                    let name = entity.get::<Name>();
                    println!("name {:?}", name);
                })
                .apply()
                .unwrap();
```

These are the entities that are being saved:

```
5v1 name Some(Name { hash: 4407845981047435273, name: "PlayerSpawn" })
4v1 name None
2v1 name Some(Name { hash: 7943398159714962848, name: "Directional Light" })
1v1 name None
12v0 name Some(Name { hash: 3891227486673895257, name: "Set" })
13v0 name None
11v0 name Some(Name { hash: 7808137298664908353, name: "Ground" })
14v0 name Some(Name { hash: 15962547213224080694, name: "Wall" })
22v1 name Some(Name { hash: 15962547213224080694, name: "Wall" })
```

But when applying, even immediately after saving the snapshot, this is what I see:

```
---------spawning entity 11v0
name Some(Name { hash: 6509597628072585063, name: "Ground" })
---------spawning entity 12v0
name Some(Name { hash: 1190878895532957759, name: "Set" })
---------spawning entity 13v0
name None
---------spawning entity 14v0
name Some(Name { hash: 12572175705635817883, name: "Wall" })
---------spawning entity 1v1
name None
---------spawning entity 19v2
name None
---------spawning entity 4v1
name None
---------spawning entity 2v2
name None
```

For some reason some entity generations have increased, and those that have no longer have `Name` components (or any components for that matter). I'm pretty sure the despawn step is deleting those entities, which I take it means they're missing from the snapshot, but I'm passing those entities to `extract_entities` so I don't understand why that would be the case.

I'm trying to figure out if there's something in my code that's causing this but in the meantime I thought maybe there's something I forgot to configure or am doing wrong? Thanks.

This is on:

```
bevy_save = "0.8.1"
bevy = "0.10.1"
```

**Comment by hankjordan** [comment]
I was able to replicate this, I am preparing a patch now. It will be released as `0.8.2`.

**Comment by frnsys** [comment]
Amazing, thanks!
